mirrord/layer/src/exec_utils.rs

```diff
@@ -269,7 +282,8 @@ pub(crate) unsafe extern "C" fn _nsget_executable_path_detour(
     let res = FN__NSGET_EXECUTABLE_PATH(path, buflen);
     if res == 0 {
         let path_buf_detour = CheckedInto::<PathBuf>::checked_into(path as *const c_char);
-        if let Bypass(FileOperationInMirrordBinTempDir(later_ptr)) = path_buf_detour {
+        if let Bypass(FileOperationInMirrordBinTempDir(path_buf)) = path_buf_detour {
+            let later_ptr: *const i8 = path_buf.as_os_str().as_encoded_bytes().as_ptr() as _;
```

Comment:
If you change what `later_ptr` is, you also have to change the code that  uses that variable in line 292 and after that, because it does not hold the information it used to hold anymore.

The calculation of `prefix_len` is now wrong. Making the `prefix_len` correct would be enough to make the rest of the function correct.

`prefix_len` is supposed to hold the length of the prefix that should be removed from the path. 

It is currently calculated as
```Rust
            let prefix_len = later_ptr.offset_from(path);
```

But after the change `later_ptr` and `path` are pointers into two different buffers, so taking their offset makes no sense.

If you really want `FileOperationInMirrordBinTempDir` to hold a `PathBuf` (which doesn't look convenient to me), then you could calculate `prefix_len` by reducing the length of `path_buf` from `*buflen`.
However I think it would be nicer to either leave the value of that variant a pointer as it was, or hold the prefix length directly (but then you would have to calculate the new pointer in the places in the code where the returned pointer was used as-is).

Reply:
I think calculating the length in a new way should be sufficient - I don't believe you can have `FileOperationInMirrordBinTempDir` contain a meaningful pointer with these changes